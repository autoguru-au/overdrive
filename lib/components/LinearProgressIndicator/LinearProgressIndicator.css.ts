import { keyframes, style } from '@vanilla-extract/css';

import { overdriveTokens as vars } from '../../themes/theme.css';

export const root = style({
	transform: 'translateZ(0)',
	transition: 'opacity 0.25s cubic-bezier(0.4, 0, 0.6, 1) 0ms',
	height: vars.space['1'],
});

const layerOneAnim = keyframes({
	'0%': {
		transform: 'translateX(0)',
		animationTimingFunction: 'cubic-bezier(0.15, 0, 0.51506, 0.40969)',
	},
	'25%': {
		transform: 'translateX(37.65191%)',
		animationTimingFunction: 'cubic-bezier(0.31033, 0.28406, 0.8, 0.73371)',
	},
	'48.35%': {
		transform: 'translateX(84.38617%)',
		animationTimingFunction: 'cubic-bezier(0.4, 0.62704, 0.6, 0.90203)',
	},
	'100%': {
		transform: 'translateX(160.27778%)',
	},
});

export const linearProgressBar = style({
	left: '-54.888891%',
	animationName: layerOneAnim,
	animationDuration: '2s',
	animationTimingFunction: 'linear',
	animationIterationCount: 'infinite',
});

const layerTwoAnim = keyframes({
	'0%': {
		transform: 'scaleX(0.08)',
		animationTimingFunction:
			'cubic-bezier(0.20503,0.05705,0.57661,0.45397)',
	},
	'19.15%': {
		transform: 'scaleX(0.4571)',
		animationTimingFunction:
			'cubic-bezier(0.15231,0.19643,0.64837,1.00432)',
	},
	'44.15%': {
		transform: 'scaleX(0.72796)',
		animationTimingFunction:
			'cubic-bezier(0.25776,-0.00316,0.21176,1.38179)',
	},
	'100%': {
		transform: 'scaleX(0.08)',
	},
});
export const linearProgressBarInner = style({
	animationName: layerTwoAnim,
	animationDuration: '2s',
	animationTimingFunction: 'linear',
	animationIterationCount: 'infinite',
});
