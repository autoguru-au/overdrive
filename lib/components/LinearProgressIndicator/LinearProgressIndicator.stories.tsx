import { Meta, StoryFn } from '@storybook/react';
import * as React from 'react';

import { LinearProgressIndicator } from '.';

export default {
	title: 'Components/Progress/Linear',
	component: LinearProgressIndicator,
	decorators: [
		(story) => (
			<div style={{ width: '100%', height: '100%' }}>{story()}</div>
		),
	],
	parameters: {
		chromatic: { disable: true },
	},
} satisfies Meta<typeof LinearProgressIndicator>;

export const standard = {};
