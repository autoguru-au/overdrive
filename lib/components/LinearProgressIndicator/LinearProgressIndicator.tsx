import clsx from 'clsx';
import React, { FunctionComponent, memo } from 'react';
import styles from './style.scss';

export interface Props {
	className?: string;
}

const LinearProgressIndicatorComponent: FunctionComponent<Props> = ({
	className = '',
}) => {
	const cls = clsx([styles.root, className]);

	return (
		<div className={cls}>
			<div className={styles.linearProgressBar}>
				<span className={styles.linearProgressBarInner} />
			</div>
		</div>
	);
};

export const LinearProgressIndicator = memo(LinearProgressIndicatorComponent);
