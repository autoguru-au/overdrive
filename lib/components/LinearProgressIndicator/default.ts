export { LinearProgressIndicator as default } from './LinearProgressIndicator';
