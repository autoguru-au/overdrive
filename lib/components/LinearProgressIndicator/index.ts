export {
	LinearProgressIndicator,
	Props as ILinearProgressIndicatorProps,
} from './LinearProgressIndicator';
