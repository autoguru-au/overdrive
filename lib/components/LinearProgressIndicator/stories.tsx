import { ComponentMeta, ComponentStory } from '@storybook/react';
import * as React from 'react';

import { LinearProgressIndicator } from '.';

export default {
	title: 'Components/Progress/Linear',
	component: LinearProgressIndicator,
	decorators: [
		(story) => (
			<div style={{ width: '100%', height: '100%' }}>{story()}</div>
		),
	],
	parameters: {
		chromatic: { disable: true },
	},
} as ComponentMeta<typeof LinearProgressIndicator>;

const Template: ComponentStory<typeof LinearProgressIndicator> = (args) => (
	<LinearProgressIndicator {...args} />
);

export const standard = Template.bind({});
