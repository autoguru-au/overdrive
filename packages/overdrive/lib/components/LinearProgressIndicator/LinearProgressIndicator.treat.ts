import { style } from 'treat';

export const root = style((theme) => ({
	height: theme.space['1'],
	transition: 'opacity 0.25s cubic-bezier(0.4, 0, 0.6, 1) 0ms',
	transform: 'translateZ(0)',
}));

export const linearProgressBar = style({
	left: '-54.888891%',
	animationDuration: '2s',
	animationTimingFunction: 'linear',
	animationIterationCount: 'infinite',

	'@keyframes': {
		'0%': {
			animationTimingFunction: 'cubic-bezier(0.15, 0, 0.51506, 0.40969)',
			transform: 'translateX(0)',
		},
		'25%': {
			animationTimingFunction:
				'cubic-bezier(0.31033, 0.28406, 0.8, 0.73371)',
			transform: 'translateX(37.65191%)',
		},
		'48.35%': {
			animationTimingFunction: 'cubic-bezier(0.4, 0.62704, 0.6, 0.90203)',
			transform: 'translateX(84.38617%)',
		},
		'100%': {
			transform: 'translateX(160.27778%)',
		},
	},
});

export const linearProgressBarInner = style({
	animationDuration: '2s',
	animationTimingFunction: 'linear',
	animationIterationCount: 'infinite',

	'@keyframes': {
		'0%': {
			animationTimingFunction:
				'cubic-bezier(0.20503,0.05705,0.57661,0.45397)',
			transform: 'scaleX(0.08)',
		},
		'19.15%': {
			animationTimingFunction:
				'cubic-bezier(0.15231,0.19643,0.64837,1.00432)',
			transform: 'scaleX(0.4571)',
		},
		'44.15%': {
			animationTimingFunction:
				'cubic-bezier(0.25776,-0.00316,0.21176,1.38179)',
			transform: 'scaleX(0.72796)',
		},
		'100%': {
			transform: 'scaleX(0.08)',
		},
	},
});
