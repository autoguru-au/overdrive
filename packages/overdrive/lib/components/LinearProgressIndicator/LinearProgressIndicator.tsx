import * as React from 'react';
import { memo, NamedExoticComponent } from 'react';
import { useStyles } from 'react-treat';

import { Box } from '../Box';
import * as styleRefs from './LinearProgressIndicator.treat';

export interface Props {
	className?: string;
}

export const LinearProgressIndicator: NamedExoticComponent<Props> = memo(
	({ className = '' }) => {
		const styles = useStyles(styleRefs);
		return (
			<Box
				position="relative"
				overflow="hidden"
				width="full"
				backgroundColour="gray200"
				className={[styles.root, className]}>
				<Box
					position="absolute"
					width="full"
					height="full"
					className={styles.linearProgressBar}>
					<Box
						is="span"
						backgroundColour="green300"
						position="absolute"
						width="full"
						height="full"
						className={styles.linearProgressBarInner}
					/>
				</Box>
			</Box>
		);
	},
);
