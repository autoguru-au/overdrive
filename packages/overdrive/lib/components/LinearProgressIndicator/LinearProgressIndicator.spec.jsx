import { render } from '@testing-library/react';
import * as React from 'react';

import { LinearProgressIndicator } from './LinearProgressIndicator';

describe('<LinearProgressIndicator />', () => {
	it('should not throw', () =>
		expect(() => render(<LinearProgressIndicator />)).not.toThrow());

	it('should match snapshot', () => {
		expect(
			render(<LinearProgressIndicator />).container.firstChild,
		).toMatchSnapshot();
	});
});
