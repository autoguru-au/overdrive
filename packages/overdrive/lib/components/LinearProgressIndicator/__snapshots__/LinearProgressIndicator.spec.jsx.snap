// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LinearProgressIndicator /> should match snapshot 1`] = `
<div
  class="base width_full position_relative overflow_hidden backgroundColours_gray200_base root_base"
>
  <div
    class="base width_full height_full position_absolute linearProgressBar"
  >
    <span
      class="base width_full height_full position_absolute backgroundColours_green300_base linearProgressBarInner"
    />
  </div>
</div>
`;
